### PR TITLE
[openbeats-3] BEATs AudioSet-2M pretraining recipe

### PR DESCRIPTION
**Pull request by jaeyeonkim99**
## What did you change?

Added the `egs2/audioset/ssl1` recipe that runs the BEATs iterative pre-training loop (from #6437) on AudioSet-2M.

- `run.sh` driving `beats.sh` over the `iter0 encoder → iter1 tokenizer → iter1 encoder` loop.
- Training configs: `conf/beats_base.yaml` (encoder), `conf/tok_beats_base.yaml` (VQ tokenizer), `conf/as2m_inf.yaml` (tokenizer inference), and the matching DeepSpeed configs.
- `local/data.sh` + `local/data_prep_as2m.py` (multiprocessing AudioSet manifest prep).
- `README.md` with setup, usage, and downstream RESULTS.
- Released the pre-trained encoder/tokenizer checkpoints on HuggingFace ([iter0](https://huggingface.co/jaeyeonkim99/BEATs-base-AS2M-iter0), [iter1](https://huggingface.co/jaeyeonkim99/BEATs-base-AS2M-iter1), [tokenizer](https://huggingface.co/jaeyeonkim99/BEATs-tokenizer-AS2M-iter1)).

---

## Why did you make this change?

The per-corpus recipe promised in #6437. It gives a runnable, reproducible AudioSet-2M BEATs pre-training recipe on top of the merged training stack, with released checkpoints and downstream numbers so the iterative pre-training can be reproduced end-to-end.

---

## Is your PR small enough?

Mostly. 22 files, ~780 lines, all recipe scripts/configs (no core code); most of the file count is standard recipe symlinks and config files that must ship together.

---

## Additional Context

* #6052
* #6436
* #6437

### CODE REVIEW

File: egs2/audioset/ssl1/conf/as2m_inf.yaml
Comment (sw005320) [inline egs2/audioset/ssl1/conf/as2m_inf.yaml:1]: Why are they separated from the other config files?
It's better to merge them to be one config file

Comment (sw005320) [reply egs2/audioset/ssl1/conf/as2m_inf.yaml:1]: Sorry, maybe I'm wrong?
Should they be separated?
In this case, can you add a comment on which tools use this config?

File: egs2/audioset/ssl1/local/log_to_tb.py
Comment (sw005320) [inline egs2/audioset/ssl1/local/log_to_tb.py:1]: This sounds a bit hucky, but if it is not easy due to the DeepSpeed trainer, we can accept this.

File: egs2/audioset/ssl1/run.sh [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    --expdir "${storage_dir}/exp" \
+    --stage 1 \
+    --stop_stage 7 \
+    --feats_type fbank \
+    --ngpu "${ngpu}" \
+    --num_nodes 1 \
+    --train_start_iter "${train_start_iter}" \
+    --train_stop_iter "${train_stop_iter}" \
+    --nj 32 \
+    --max_wav_duration 11 \
```
Comment (sw005320) [inline egs2/audioset/ssl1/run.sh:55]: What's this?
Can you explain it?

**Review by sw005320 — COMMENTED** [review]
- There are multiple config files, and it looks complicated
  - Can you merge some of them?
  - Can you rename them and add a comment to the header so that the users can understand them more clearly?
- Please fix the CI errors

**Comment by sw005320** [comment]
Thanks.
Please fix the CI error.

@popcornell and @Shikhar-S, can you check this PR?
I think most parts look good to me.

File: egs2/audioset/ssl1/local/log_to_tb.py
Comment (jaeyeonkim99) [reply egs2/audioset/ssl1/local/log_to_tb.py:1]: This was just a post-hoc helper I ran by hand to view `train.log` in TensorBoard, but mistakenly included in the commit. Removed it. Fixed in [d0190f4](https://github.com/espnet/espnet/pull/6498/commits/d0190f44f3ad85a2b6b63147a2747662ec0fd445).

File: egs2/audioset/ssl1/run.sh (hunk 1 above)
Comment (jaeyeonkim99) [reply egs2/audioset/ssl1/run.sh:55]: `--max_wav_duration 11` filters out training clips longer than 11s during data filtering. AudioSet clips are nominally 10s, so the 1s margin keeps the genuine ~10s > clips while dropping a handful of  malformed/overlong files.

**Comment by jaeyeonkim99** [comment]
> * There are multiple config files, and it looks complicated
>   
>   * Can you merge some of them?
>   * Can you rename them and add a comment to the header so that the users can understand them more clearly?
> * Please fix the CI errors

Each one maps 1:1 to a distinct `beats.sh` stage/argument, so they can't be easily merged into fewer files:
- `conf/beats_base.yaml` — encoder training, stage 7 (`--train_config`)
- `conf/tok_beats_base.yaml` — tokenizer training, stage 7 (`--tokenizer_train_config`)
- `conf/as2m_inf.yaml` — tokenizer inference, stage 5 (`--tokenizer_inference_config`). Has distinct values per dataset if training is happening over multiple dataset.
- `conf/ds_beats.json`, `conf/ds_beats_tok.json` — the DeepSpeed configs referenced by
   the two training YAMLs. Kept separate because DeepSpeed requires its own JSON per run,
   and the encoder/tokenizer use different LR schedules (5e-4 vs 5e-5).
- `conf/{slurm,queue,pbs}.conf` — the standard recipe cluster configs.

I added config→stage/argument table to the recipe README. Fixed in [d0190f4](https://github.com/espnet/espnet/pull/6498/commits/d0190f44f3ad85a2b6b63147a2747662ec0fd445).

File: egs2/audioset/ssl1/conf/as2m_inf.yaml
Comment (jaeyeonkim99) [reply egs2/audioset/ssl1/conf/as2m_inf.yaml:1]: `conf/as2m_inf.yaml` configures the tokenizer *inference* step (stage 5), where the trained VQ tokenizer encodes the corpus into the discrete tokens used as targets for iter>0 encoder training. It's kept separate from the training configs because inference runs per source dataset with its own fbank normalization stats. In the multi-dataset, the original OpenBEATs implementation has onee `*_inf.yaml` per dataset, decoupled from the single training config.  AudioSet-2M is a single dataset so there's just one file here, but the recipe keeps the same structure. I added a header comment on which stage uses it and a> config→stage/argument table to the recipe README. Fixed in [d0190f4](https://github.com/espnet/espnet/pull/6498/commits/d0190f44f3ad85a2b6b63147a2747662ec0fd445).

**Comment by jaeyeonkim99** [comment]
> Thanks. Please fix the CI error.
> 
> @popcornell and @Shikhar-S, can you check this PR? I think most parts look good to me.

Fixed the CI error in [f9d78cd](https://github.com/espnet/espnet/pull/6498/commits/f9d78cd8b8dbdc4cbac151b600ddf9cea7a8776e)

File: egs2/audioset/ssl1/run.sh (hunk 1 above)
Comment (sw005320) [reply egs2/audioset/ssl1/run.sh:55]: Make sense.
Can you add this comment to this file, then?

**Comment by sw005320** [comment]
@popcornell and @Shikhar-S, this is a reminder.
The PR is in good shape now.

**Review by sw005320 — APPROVED** [review]

**Comment by popcornell** [comment]
On it, sorry about missing it

**Comment by Shikhar-S** [comment]
> @popcornell and @Shikhar-S, this is a reminder. The PR is in good shape now.

Sorry I missed this email thread. ~Working on it.~
Edit: Yes, looks good to me! Thanks @jaeyeonkim99

**Comment by popcornell** [comment]
I also think we can merge on my side too. 
Last week there was I think one possible issue but i did a commit

**Comment by sw005320** [comment]
OK, thanks!
This will be merged after the CI check

**Comment by sw005320** [comment]
Thanks!
